stylus-core/src/lib.rs

```diff
@@ -7,4 +7,20 @@ pub mod deploy;
 pub mod host;
 pub mod storage;
 
+use alloy_sol_types::{abi::token::WordToken, SolEvent, TopicList};
 pub use host::*;
+
+/// Emits a typed, Alloy log.
+pub fn log<T: SolEvent>(vm: &dyn Host, event: T) {
```

Comment:
Could this be a trait method so we can call it as `vm().log(event)`?

Reply:
Unfortunately not because it would prevent us from constructing a box dyn Host, as it now has a generic param. Even if we just add the generic to the trait method, it triggers a bunch of errors that don't have an easy solution for now :(